Source/JNA/waffle-tomcat85/src/main/java/waffle/apache/NegotiateAuthenticator.java

```diff
@@ -158,12 +159,11 @@ public boolean authenticate(final Request request, final HttpServletResponse res
             try {
                 this.log.debug("logged in user: {} ({})", windowsIdentity.getFqn(), windowsIdentity.getSidString());
 
-                final GenericWindowsPrincipal windowsPrincipal = new GenericWindowsPrincipal(windowsIdentity,
-                        this.principalFormat, this.roleFormat);
+                final GenericPrincipal genericPrincipal = createPrincipal(windowsIdentity);
 
-                this.log.debug("roles: {}", windowsPrincipal.getRolesString());
+                this.log.debug("roles: {}",  String.join(", ", genericPrincipal.getRoles()));
```

Comment:
Please revert the change above.  getRolesString() already does the string joiner.

Reply:
nevermind I see why now.

Reply:
Using the toString-Method log output would give the possibility to use more generic interfaces
this.log.debug("roles: {}",  String.join(", ", genericPrincipal.getRoles()));
->
this.log.debug("principal: {}",  principal.toString());